ddtrace/propagation/http.py

```diff
@@ -323,7 +323,8 @@ def _extract(headers):
         if sampling_priority == USER_KEEP:
             if not meta:
                 meta = {}
-            meta[SAMPLING_DECISION_TRACE_TAG_KEY] = f"-{SamplingMechanism.TRACE_SAMPLING_RULE}"
+            if not meta.get(SAMPLING_DECISION_TRACE_TAG_KEY):
+                meta[SAMPLING_DECISION_TRACE_TAG_KEY] = f"-{SamplingMechanism.TRACE_SAMPLING_RULE}"
```

Comment:
We don't technically know the mechanism, so we're taking a best guess here? For example it could've originally been `SamplingMechanism.MANUAL` right?

Reply:
I guess we were already doing that though, so this just makes it far less often that we guess.

Reply:
Yes, that's right. It's also not exactly a "guess" - instead it's the new expectation for this specific case set out in the RFC, for which there will eventually be a system-test.